config/settings: Add microshift preset to helper

praveenkumar opened the pull request:
**Fixes:** Issue #3576

cfergeau commented on pkg/crc/config/settings.go:
```diff
@@ -70,7 +70,7 @@ func RegisterSettings(cfg *Config) {
 	// Preset setting should be on top because CPUs/Memory config depend on it.
 	cfg.AddSetting(Preset, string(preset.OpenShift), validatePreset, RequiresDeleteAndSetupMsg,
-		fmt.Sprintf("Virtual machine preset (valid values are: %s, %s and %s)", preset.Podman, preset.OpenShift, preset.OKD))
+		fmt.Sprintf("Virtual machine preset (valid values are: %s, %s, %s and %s)", preset.Podman, preset.OpenShift, preset.Microshift, preset.OKD))
```
A more future-proof alternative is to add `func preset.AllPresets() []preset.Preset` and use this in `RegisterSettings` and `validatePreset`

praveenkumar committed:
Preset: Add presetMap variable

It remove some of the duplication we are having with switch case
and also allow us to use it for another helper which can provide
us list of all preset ( next commit)

praveenkumar committed:
preset: Add AllPresets helper

It will help us to not have update presets at different place. In
this pr it is consumed by preset setting and validation.

cfergeau commented on pkg/crc/preset/preset.go:
```diff
@@ -15,16 +15,20 @@ const (
+		if pSet == preset {
+			return pString
+		}
 	}
```
you have a map, you don't need to iterate over it
```
presetStr, presetExists := presetMap()[preset]
if !presetExists {
    return "invalid"
}
return presetStr
```

praveenkumar replied:
updated

cfergeau commented on pkg/crc/preset/preset.go:
```diff
@@ -15,16 +15,20 @@ const (
+		OKD:        string(OKD),
+		Microshift: string(Microshift),
+	}
+}
```
```
var presetMap = map[Preset]string {
...
}
```
should be enough, and allow us to create the map only once?

praveenkumar replied:
updated.

cfergeau commented on pkg/crc/config/settings.go:
```diff
@@ -70,7 +70,7 @@ func RegisterSettings(cfg *Config) {
 	// Preset setting should be on top because CPUs/Memory config depend on it.
 	cfg.AddSetting(Preset, string(preset.OpenShift), validatePreset, RequiresDeleteAndSetupMsg,
-		fmt.Sprintf("Virtual machine preset (valid values are: %s, %s and %s)", preset.Podman, preset.OpenShift, preset.OKD))
+		fmt.Sprintf("Virtual machine preset (valid values are: %s)", preset.AllPresets()))
```
What does the string looks like ? I don't know how go converts an array to a string.

praveenkumar replied:
@cfergeau it have `Virtual machine preset (valid values are: [openshift okd podman microshift])` if we really want to have as string then another helper is required.

cfergeau replied:
`strings.Join(preset.AllPresets, ", ")` might work.

praveenkumar replied:
> strings.Join(preset.AllPresets, ", ") might work

No, beacuse it complain `preset.AllPresets` is not `[]string` type so not able to use.

cfergeau replied:
Let's say `[openshift okd podman microshift]` is good enough if you don't want the extra code.
